crates/topos-p2p/src/network.rs

```diff
@@ -171,6 +171,21 @@ impl<'a> NetworkBuilder<'a> {
 
         let grpc_over_p2p = GrpcOverP2P::new(command_sender.clone());
 
+        let listen_addr = self
+            .listen_addresses
+            .take()
+            .expect("Node requires at least one address to listen for incoming connections");
+
+        let public_addresses = if let Some(addresses) = self.public_addresses.take() {
+            if addresses.is_empty() {
+                listen_addr.clone()
+            } else {
+                addresses
+            }
+        } else {
+            listen_addr.clone()
+        };
```

Comment:
maybe something in the likes:

```suggestion
        let public_addresses = match self.public_addresses.take() {
            None | Some(addresses) if addresses.is_empty() => listen_addr.clone()
            Some(addresses) => addresses
        };
```

might be easier to follow what is what

Reply:
This will not compile as `addresses` is not present in the `None`...

Reply:
My bad, but you got the idea I guess, just need to separate the `None` case then no? Nitpick but it feels clearer